SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialVirtualWorkerTranslator);
+
+namespace {
+	// The translation schema is a list of Mappings, where each entry has a virtual and physical worker ID. 
+	inline void GetMappingFromSchema(Schema_Object* Object, Schema_FieldId Id, VirtualToPhysicalWorkerMap& Mapping)
+	{
+		int32 TranslationCount = (int32)Schema_GetObjectCount(Object, 1);
+		Mapping.Reserve(TranslationCount);
+
+		for (int32 i = 0; i < TranslationCount; i++)
+		{
+			// Get each entry of the list and then unpack the virtual and physical IDs from the entry.
+			Schema_Object* MappingObject = Schema_IndexObject(Object, SpatialConstants::TRANSLATION_VIRTUAL_WORKER_MAPPING_ID, i);
```

Comment:
I went and checked what the schema object is and I see why I was confused here. We've used a list of pairs instead of a map. Both of which have the same internal representation.
This raises the question of if we should just use a map in the schema file.

Reply:
I thought about using a map  (especially when I discovered what a pain it was to pack/unpack the schema)  but I convinced myself that there might one day be a world where we want two physical workers to simulate the same virtual worker.   (for instance, if we incorporate offloading in somehow)

Reply:
I don't think that can be made to work. You can have one virtual worker dealing with multiple authority intents, but if we say that a virtual worker can be simulated by multiple physical workers then we need to remake the same abstraction again inside the concept of a virtual worker but with no gain in flexibility.

Reply:
Also all runtime proposals for user space load balancing have a bijection between virtual workers and physical workers.

Reply:
Although it's worth noting - I think the packing and unpacking is identical in both cases as our maps our just a list of pairs (at least for now).

Reply:
I was actually thinking of it as poor man's hysteresis. If we had 3 intents, two big ones and one "border" intent, we could have the border intent be valid on both of 2 physical workers. Now that I'm thinking about it, the zone/Strategy is probably the better place to do that. 

At the end of the day, though I don't really care if it's a map or a list of pairs. I'm inclined to stick with this because they seem identical and this is already written, but if you feel strongly, I can change it.